server/events.go

```diff
@@ -767,6 +807,31 @@ func (s *Server) statszReq(sub *subscription, _ *client, subject, reply string,
 	s.sendStatsz(reply)
 }
 
+func (s *Server) zReq(reply string, msg []byte, optz interface{}, respf func() (interface{}, error)) {
+	if !s.EventsEnabled() || reply == _EMPTY_ {
+		return
+	}
+	server := &ServerInfo{}
+	response := map[string]interface{}{"server": server}
+	var err error
+	status := 0
+	if len(msg) != 0 {
+		err = json.Unmarshal(msg, optz)
+		status = http.StatusBadRequest // status is only included on error, so record how far execution got
+	}
+	if err == nil {
+		response["data"], err = respf()
+		status = http.StatusInternalServerError
```

Comment:
This seem weird to me. You set it regardless of the `err` result.

Reply:
because status is only returned when there is an error, I can basically set it all the time essentially keeping track of how far I got. But will change

Reply:
Right, hence comment above when doing the unmarshal.